### PR TITLE
[WIP] Support Unnormalized inverse FFT

### PR DESCRIPTION
**Pull request by mattvend**
This pull request aims at fixing #944 

With the `"unnormalized"` mode, FFTs become unnormalized in both directions. This is a deviation from the numpy API.

### Done
- added an extra normalization mode `"unnormalize"` on top of `None` and `"ortho"` for FFTs
- tests updated
- documentation

### Not done yet
- ihfft not supported yet. I can not get it to work, I struggle to fix it, there is something wrong there.
- Should the test suite of `cupyx.scipy.fft` and legacy `cupyx.scipy.fftpack` be updated ?

### to discuss
`"unnormalize"` can be changed to `"unnormalized"` if preferred.

### CODE REVIEW

**Comment by asi1024** [comment]
@mattvend Thank you for the PR! The interface of CuPy is designed to obey that of NumPy basically, so we will start the review after NumPy supports this feature.

**Comment by leofang** [comment]
> The interface of CuPy is designed to obey that of NumPy basically, so we will start the review after NumPy supports this feature.

My two cents: I don't think for this particular request it has to be the blocker, since the scope is well-defined and not interfering with normal usage or backward compatibility. I would suggest mark it clearly as "experimental", though, so that we leave room for (unlikely) future changes.

That said, I don't think it is too hard to fix this in upstream (https://github.com/numpy/numpy/issues/16126) first.

**Comment by leofang** [comment]
@mattvend FYI numpy/numpy#16476 is in!

**Comment by leofang** [comment]
ref: scipy/scipy#12532 I can handle `cupyx.scipy.fft` later.

**Comment by asi1024** [comment]
numpy/numpy#16476 will be reflected in NumPy 1.20 and scipy/scipy#12532 in SciPy 1.6. Let's support them in `cupy.fft` and `cupyx.scipy.fft` after stable NumPy and SciPy of the version, respectively.

File: cupy/fft/fft.py [hunk 1]
```diff
@@ -542,7 +544,8 @@ def fft(a, n=None, axis=-1, norm=None):
             is not given, the length of the input along the axis specified by
             ``axis`` is used.
         axis (int): Axis over which to compute the FFT.
-        norm (None or ``"ortho"``): Keyword to specify the normalization mode.
+        norm (None or ``"ortho"`` or ``"unnormalize"``): Keyword to specify
```
Comment (asi1024) [inline cupy/fft/fft.py:547]: Could you fix to take "forward" or "backward" as NumPy does?

Comment (mattvend) [reply cupy/fft/fft.py:547]: Hi all. I still plan to work on this, but got busy at work those days, I will do my best in the coming weeks.

Comment (asi1024) [reply cupy/fft/fft.py:547]: Any updates?

**Comment by leofang** [comment]
Hi @mattvend Do you need help with or someone to take over this PR? I am happy to help either way. Let us know!

**Comment by mattvend** [comment]
Hi @leofang Just resumed work on this pr. I synced my branch with master, hence this commit. I will align my work on the numpy changes, so that API are compatible. I hope to do that in the coming days.

**Comment by leofang** [comment]
Hi @mattvend Friendly nudge 🙂 Any chance you have time to work on this? The "feature complete" time for rc1 is by the end of this month, and personally I'd like to see this in. If you don't have time then I might be able to take over and finish it.

**Comment by mattvend** [comment]
Hi @leofang 
Please go ahead with it. I will probably not have time to finish it. Please be my guest and terminate. I can't unfortunately complete this work.

**Pull request closed without merging**